### PR TITLE
complete equipment pr

### PR DESCRIPTION
**Pull request by wolyslager**
image
image
image

### CODE REVIEW

File: components/states/DiveShopConfigFetchState.jsx [hunk 1]
```diff
@@ -20,6 +20,9 @@ const DiveShopConfigFetchState = () => {
     if (url_hash) {
       fetchData(url_hash);
     }
+    send({
+      type: STATE_ACTIONS.FETCH_SUCCESS,
+    });
```
Comment (adolfoportilla) [inline components/states/DiveShopConfigFetchState.jsx:25]: nope, should not be sent here!

File: components/states/EquipmentState.jsx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  const [maskChoice, setMaskChoice] = React.useState("");
+
+  const [tankSelectVisible, setTankSelectVisible] = React.useState(false);
+  const [tankSize, setTankSize] = React.useState("");
+
+  return (
+    <StatePage>
+      <StateTitle title="Select the equipment you will need" />
+      <StateCards>
+        <FinsCard
```
Comment (adolfoportilla) [inline components/states/EquipmentState.jsx:44]: can the select visibility be inside the component, and not here?

File: components/states/EquipmentState.jsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          onClick={() =>
+            send({
+              type: STATE_ACTIONS.DIVER_INFORMATION,
+              value: {
+                finSize,
+                bcdSize,
+                wetsuitSize,
+                regulatorChoice,
+                maskChoice,
+                tankSize,
```
Comment (adolfoportilla) [inline components/states/EquipmentState.jsx:96]: This is ok for now, but we will need to change this if we want to have some that are required and some that are not

File: components/states/equipment/BCDCard/index.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        <CardComponent
+          icon={
+            <SelectComponent
+              setValue={setBcdSize}
+              label="Size"
+              options={sizes}
+            />
+          }
+          text="Select Size"
+          onClick={() => setBcdSelectVisible(!bcdSelectVisible)}
```
Comment (adolfoportilla) [inline components/states/equipment/BCDCard/index.jsx:40]: All of these icons are a copy of each other, with only a couple of differences. I will say, lets make a component for all of them and use it in all?

I think we can generalize the pattern

File: utils/state-machine.js [hunk 5]
```diff
@@ -99,6 +108,7 @@ export const STATE_ACTIONS = {
   DIVER_NOT_CERTIFIED: "DIVER_NOT_CERTIFIED",
   FETCH_SUCCESS: "FETCH_SUCCESS",
   CREATE_SUPABASE_RESERVATION: "CREATE_SUPABASE_RESERVATION",
+  FIN_SIZE: "FIN_SIZE",
```
Comment (adolfoportilla) [inline utils/state-machine.js:111]: why fin size, and equipment?

File: utils/state-machine.js [hunk 6]
```diff
@@ -254,27 +265,55 @@ export const reservationMachine = createMachine(
...
         },
       },
+      equipment: {
+        on: {
+          [STATE_ACTIONS.DIVER_INFORMATION]: {
+            target: "diverInformation",
+            actions: ["pushToPreviousState", "setEquipment"],
+          },
+
+          [STATE_ACTIONS.FIN_SIZE]: {
```
Comment (adolfoportilla) [inline utils/state-machine.js:286]: I don't think we shouuld have it this way, ideally equipment + fin size is one state

**Review by adolfoportilla — CHANGES_REQUESTED** [review]
looking good, but seems to much code copy past, and the state ideally is just one, vs 2

File: components/states/DiveShopConfigFetchState.jsx (hunk 1 above)
Comment (wolyslager) [reply components/states/DiveShopConfigFetchState.jsx:25]: Did it so I can test locally - forgot to remove

**Review by adolfoportilla — APPROVED** [review]
Butter to my eyes

**Pull request merged**